### PR TITLE
Introduces documentation for the docker image

### PR DESCRIPTION
**Pull request by kelsos**

### CODE REVIEW

File: docs/installation_guide.rst [hunk 1]
```diff
@@ -108,6 +108,116 @@ If you get "The python backend crashed" or any other error please run the execut
...
+
+   Versions up to v1.13.2 report a dev version instead of the actual
+   release. This is due to an issue during the build process.
+
+You can find all the available docker images at `DockerHub`_.
+
+   !Warning: It is important to keep in mind that is advisable to run
+   the docker image in a secure environment. Such an environment would
+   be for example running the docker image in the host machine and
+   accessing it only via NAT. Under any circumstances you should not
```
Comment (LefterisJP) [inline docs/installation_guide.rst:125]: Under any circumstances you should not -> Under no circumstances should you

File: docs/installation_guide.rst [hunk 2]
```diff
@@ -108,6 +108,116 @@ If you get "The python backend crashed" or any other error please run the execut
...
+To start a container based on the latest rotki image you can run the
+following::
+
+   docker run -d --name rotki \
+       -v $HOME/.rotki/data:/data \
+       -v $HOME/.rotki/logs:/logs \
+       rotki/rotki:latest
+
+This will start a new container that stores the data and logs into a
+``.rotki`` directory under the user’s home directory. Will will be able
```
Comment (LefterisJP) [inline docs/installation_guide.rst:144]: Will will -> You will

File: docs/installation_guide.rst [hunk 3]
```diff
@@ -108,6 +108,116 @@ If you get "The python backend crashed" or any other error please run the execut
...
+create a new container::
+
+   docker run -d --name rotki \
+       -v $HOME/.rotki/data:/data \
+       -v $HOME/.rotki/logs:/logs \
+       rotki/rotki:latest
+
+Please ensure that the volumes from the previous container are used
+again. If you don’t use the same volumes, then your old accounts and
+date will be unavailable.
```
Comment (LefterisJP) [inline docs/installation_guide.rst:190]: date -> data

File: docs/installation_guide.rst [hunk 4]
```diff
@@ -108,6 +108,116 @@ If you get "The python backend crashed" or any other error please run the execut
...
+
+.. _DockerHub: https://hub.docker.com/r/rotki/rotki/
+
+Moving the accounts from the app
```
Comment (LefterisJP) [inline docs/installation_guide.rst:194]: suggestion: desktop app (?)

File: docs/installation_guide.rst [hunk 5]
```diff
@@ -108,6 +108,116 @@ If you get "The python backend crashed" or any other error please run the execut
...
+--------------------------------
+
+If for any reason you would like to move your accounts from the rotki application to
+docker you can do so by copying the data directory of the application to the docker
+data volume.
+
+You can find where the rotki application data is stored in the :ref:`rotki_data_directory`.
+
+To move your existing accounts from the application, you can copy the contents of
+application data directory to the data volume mount point.
```
Comment (LefterisJP) [inline docs/installation_guide.rst:204]: of application -> of the application

**Review by LefterisJP — APPROVED** [review]
Thanks for the docs. Looks good. Left some typos fixes comments.

**Comment by CruiseMan** [comment]
@kelsos Thank you in dded, i literally had a dream last night to add Docker docs. Could you also add docs on how to build Docker image from source to here https://rotki.readthedocs.io/en/latest/installation_guide.html#build-from-source ?

**Comment by kelsos** [comment]
> @kelsos Thank you indeed, i literally had a dream last night to add Docker docs. Could you also add docs on how to build Docker image from source to here https://rotki.readthedocs.io/en/latest/installation_guide.html#build-from-source ?

sure that should be way easier, let me open an issue.

**Comment by CruiseMan** [comment]
> > @kelsos Thank you indeed, i literally had a dream last night to add Docker docs. Could you also add docs on how to build Docker image from source to here https://rotki.readthedocs.io/en/latest/installation_guide.html#build-from-source ?
> 
> sure that should be way easier, let me open an issue.

After i run docker commend, i'm not able to reach rotki from http://172.17.0.2/ is port `80` should be showed here as usually docker shoe exposed port(s) ? Did it ever worked for you ? I checked `nginx` config look ok, and also notice ports supplyed inside docker for back end is `4242` https://github.com/rotki/rotki/blob/0775cbd54b3449b12088cb8a05a1a5c2a0e47405/packaging/docker/docker-entrypoint.sh#L3 but in docs rotki shows `5042` https://rotki.readthedocs.io/en/latest/api.html#get--api-(version)-users 
Screen Shot 2021-02-10 at 10 31 48 PM

**Comment by CruiseMan** [comment]
Still learning codebase and build process

**Comment by kelsos** [comment]
Hey @CruiseMan I did test everything on my machine while making the guide. And everything worked fine.

You can change the port to whatever you want. The documentation has `5042` as an example but the frontend uses `4242` even in the application.

However, this doesn't matter in the case of the docker container. this backend port is not directly reachable outside the container.
To access the backend in docker you just do the `/api/` requests at the same port `80` as you would do with the frontend.

The nginx configuration will proxy any requests to the `api` to the internal `4242` port making it an implementation detail.

Also, this might be a stupid question to ask, but was 172.17.0.2 the container IP returned by the command or the one that I mentioned in the example? 

Any more information you could provide would really help. What host OS did you use, maybe there is something that doesn't work on all OSes.

**Comment by CruiseMan** [comment]
> Hey @CruiseMan I did test everything on my machine while making the guide. And everything worked fine.
> 
> You can change the port to whatever you want. The documentation has `5042` as an example but the frontend uses `4242` even in the application.
> 
> However, this doesn't matter in the case of the docker container. this backend port is not directly reachable outside the container.
> To access the backend in docker you just do the `/api/` requests at the same port `80` as you would do with the frontend.
> 
> The nginx configuration will proxy any requests to the `api` to the internal `4242` port making it an implementation detail.
> 
> Also, this might be a stupid question to ask, but was 172.17.0.2 the container IP returned by the command or the one that I mentioned in the example?
> 
> Any more information you could provide would really help. What host OS did you use, maybe there is something that doesn't work on all OSes.

Yehh right, shroud work. MacOs Big Sure 11.2.1
Screen Shot 2021-02-11 at 1 33 44 PM

**Comment by CruiseMan** [comment]
@kelsos I'm running pi-hole on RP4 and ideally want to run rotki, but they both listen on port `80`, how would you make them run on one machine ?

**Comment by CruiseMan** [comment]
`docker inspect rotki`
```json
[
    {
        "Id": "f2ab3a320a9e6c1ec9cdea325689548396af2ceec571b5957e1f099c959580a0",
        "Created": "2021-02-11T21:37:38.966724848Z",
        "Path": "/docker-entrypoint.sh",
        "Args": [
            "sh",
            "-c",
            "exec /opt/rotki/docker-entrypoint.sh"
        ],
        "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2021-02-11T21:37:40.70951068Z",
            "FinishedAt": "2021-02-11T21:37:40.708886946Z",
            "Health": {
                "Status": "unhealthy",
                "FailingStreak": 0,
                "Log": []
            }
        },
        "Image": "sha256:cca33bca8ae601269b82c1da0d6ee26d22ba97dcf4b41bb0c8dbc72c949592e5",
        "ResolvConfPath": "/var/lib/docker/containers/f2ab3a320a9e6c1ec9cdea325689548396af2ceec571b5957e1f099c959580a0/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/f2ab3a320a9e6c1ec9cdea325689548396af2ceec571b5957e1f099c959580a0/hostname",
        "HostsPath": "/var/lib/docker/containers/f2ab3a320a9e6c1ec9cdea325689548396af2ceec571b5957e1f099c959580a0/hosts",
        "LogPath": "/var/lib/docker/containers/f2ab3a320a9e6c1ec9cdea325689548396af2ceec571b5957e1f099c959580a0/f2ab3a320a9e6c1ec9cdea325689548396af2ceec571b5957e1f099c959580a0-json.log",
        "Name": "/rotki",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/home/pi/.rotki/data:/data",
                "/home/pi/.rotki/logs:/logs"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": null,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/bd8192309988d49fd65dcfaf412984681c36f1086ce9f9d334da32dc301bae51-init/diff:/var/lib/docker/overlay2/7641c4503c5509384fc5eed356f6a8d5b5221b36b14a4aca3f1c563abccd290d/diff:/var/lib/docker/overlay2/5b70e92bc3991a6f220e4dde3a8a5a03a97d021923f1ef618ed9bdadfd7143dd/diff:/var/lib/docker/overlay2/a444e130ba1e021a8164275c467e57c59173b10e7069efb158b2f08e40ad4721/diff:/var/lib/docker/overlay2/4f79ddf31e8787a141cdb9b1bd0ebfcb85da121ad7b375ba7b2cc7bb42075bfe/diff:/var/lib/docker/overlay2/f9d42a99dfc53a021bbb4c5b646db22e27b20920594de651163dabf68777d5b3/diff:/var/lib/docker/overlay2/00d7c6aaf3ac831bc80ae91c1ea047de2979bf4098a089576b3cc2ae4133fb11/diff:/var/lib/docker/overlay2/55f7fce46b5a15ccca94c9acafb60fda72e79af236580b8569b8eee9455dbe7a/diff:/var/lib/docker/overlay2/e9b9f915dd91c0b7f4292c18a3857d59a3540bb7c011791c3dc813c1870f70fc/diff:/var/lib/docker/overlay2/9b76500d1dfca4f44f3ed10cd752230431ab19e51246c68b7cfbb3ac9eb16878/diff:/var/lib/docker/overlay2/cc6c3c8bab9b6a88c9792e622e9fde1490e5d67a708cddfd7963dd0de9a7f0b8/diff:/var/lib/docker/overlay2/d690a6f7d81cf4070bbaa692a6e4af504dddcfe0fe09990035067bd2f2498731/diff",
                "MergedDir": "/var/lib/docker/overlay2/bd8192309988d49fd65dcfaf412984681c36f1086ce9f9d334da32dc301bae51/merged",
                "UpperDir": "/var/lib/docker/overlay2/bd8192309988d49fd65dcfaf412984681c36f1086ce9f9d334da32dc301bae51/diff",
                "WorkDir": "/var/lib/docker/overlay2/bd8192309988d49fd65dcfaf412984681c36f1086ce9f9d334da32dc301bae51/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/home/pi/.rotki/data",
                "Destination": "/data",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/home/pi/.rotki/logs",
                "Destination": "/logs",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "f2ab3a320a9e",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "80/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "NGINX_VERSION=1.18.0",
                "NJS_VERSION=0.4.4",
                "PKG_RELEASE=2~buster",
                "REVISION=2a4369d8d7e51443190c40de3ecbe1fefdac194c",
                "ROTKI_VERSION=v1.13.3"
            ],
            "Cmd": [
                "sh",
                "-c",
                "exec /opt/rotki/docker-entrypoint.sh"
            ],
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "curl --fail http://localhost/api/1/users || exit 1"
                ]
            },
            "Image": "rotki/rotki:latest",
            "Volumes": {
                "/data": {},
                "/logs": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "/docker-entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "maintainer": "Rotki Solutions GmbH <info@rotki.com>"
            },
            "StopSignal": "SIGQUIT"
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "d21415acfca73a34aa46d740936cd454a904036cae55b20d0f9473cd7909cd26",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/d21415acfca7",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "2756a63b7837daaa8cc8550bcca2ef504db9a59ac2d9a6e08e31220b3632bbcc",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Comment by kelsos** [comment]
Ok so you are actually running the image on the RPI?

**Comment by CruiseMan** [comment]
> Ok so you are actually running the image on the RPI?

I tried to run but it exited due to pihole running on port 80 i guess

**Comment by kelsos** [comment]
At least inspect says that there is a problem, and you could check the log folder for any logs.

The image is build for x86_64 so that might also be a problem. I need to check how easy would be to build for aarch64 since running on an RPI is something I also considered doing.

to bind to a different port https://docs.docker.com/config/containers/container-networking/#published-ports you can check this.

**Comment by CruiseMan** [comment]
No logs in folder founded. 
Screen Shot 2021-02-11 at 1 52 18 PM

Do you think make sense to build image with something like `-p 172.17.0.2:8084:80` port 8084 chosen as an example to avoid port 80 conflict ?

**Comment by kelsos** [comment]
You can pass this `-p` when you create the container if you want to map to localhost. so you can do a `localhost:8084` for example instead of 172.17.0.2:80. But like I mentioned before I am not sure how rpi handles images for a different architecture. the problem might not be the port number in the rpi's case.

**Comment by CruiseMan** [comment]
> You can pass this `-p` when you create the container if you want to map to localhost. so you can do a `localhost:8084` for example instead of 172.17.0.2:80. But like I mentioned before I am not sure how rpi handles images for a different architecture. the problem might not be the port number in the rpi's case.

haha it worked:
`
docker run -d --name rotki \
       -p 8084:80 \
       -v $HOME/.rotki/data:/data \
       -v $HOME/.rotki/logs:/logs \
       rotki/rotki:latest
`

Screen Shot 2021-02-11 at 2 12 25 PM


`http://172.17.0.2:8084/` doesn't work

**Comment by CruiseMan** [comment]
I think make sense to map 8084 to 80 to avoid common port `80` conflict

**Comment by kelsos** [comment]
I am glad it worked, btw @CruiseMan do you run rotki on docker on RPI and you try to access it from a different computer in the same network?

**Comment by CruiseMan** [comment]
> I am glad it worked, btw @CruiseMan do you run rotki on docker on RPI and you try to access it from a different computer in the same network?

I'm/it failed to start on RP4, so running on macOS

**Pull request merged**